### PR TITLE
Add ImageParam::trace_loads

### PR DESCRIPTION
**Pull request by steven-johnson**
Previously, you could only trace loads on an ImageParam by enabling trace_loads globally. This tweaks the trace code to allow it on a per-ImageParam basis.

### CODE REVIEW

File: src/Tracing.cpp [hunk 1]
```diff
@@ -72,6 +72,18 @@ class InjectTracing : public IRMutator2 {
             trace_parent = Variable::make(Int(32), op->name + ".trace_id");
         } else if (op->call_type == Call::Image) {
             trace_it = trace_all_loads;
+            // If there is a Function in the env named "name_im", assume that
```
Comment (abadams) [inline src/Tracing.cpp]: name_im could be scheduled and have actual loads from it (via ImageParam::in). I think we should only trace loads from the ImageParam if name_im is scheduled inline.

Comment (steven-johnson) [reply src/Tracing.cpp]: Fixed

**Comment by steven-johnson** [comment]
Monday Morning Review Ping

File: test/correctness/tracing.cpp [hunk 2]
```diff
@@ -132,6 +133,13 @@ int main(int argc, char **argv) {
     g.store_root().compute_at(f, x);
     g.trace_stores().trace_loads().trace_realizations();
 
+    input.trace_loads();
+
+    // Make it all-zero so the existing output values are the same
```
Comment (abadams) [inline test/correctness/tracing.cpp]: By "the same" I'm guessing you mean the same as they were before you made this change, but that won't be apparent to anyone reading this later. Maybe just cut the comment.

**Pull request merged**